Sources/YogaKit/YGLayout.mm

```diff
@@ -227,11 +227,13 @@ - (void)markDirty {
   // the measure function. Since we already know that this is a leaf,
   // this *should* be fine. Forgive me Hack Gods.
   const YGNodeRef node = self.node;
-  if (!YGNodeHasMeasureFunc(node)) {
+  if (!YGNodeHasMeasureFunc(node) && self.numberOfChildren == 0) {
     YGNodeSetMeasureFunc(node, YGMeasureView);
   }
 
-  YGNodeMarkDirty(node);
+  if (YGNodeHasMeasureFunc(node)) {
+    YGNodeMarkDirty(node);
+  }
```

Comment:
I considered modifying the isLeaf logic in YGLayout, but decided not to proceed because there are separate properties being used, making it difficult to guarantee the existing logic.

Therefore, add defensive logic for the assertFatal conditions inside Yoga.